django/utils/timezone.py

```diff
@@ -249,3 +248,29 @@ def make_naive(value, timezone=None):
     if is_naive(value):
         raise ValueError("make_naive() cannot be applied to a naive datetime")
     return value.astimezone(timezone).replace(tzinfo=None)
+
+
+_PYTZ_BASE_CLASSES = None
+
+
+def _is_pytz_zone(tz):
+    """Checks if a zone is a pytz zone."""
+    global _PYTZ_BASE_CLASSES
+    if _PYTZ_BASE_CLASSES is None:
+        _PYTZ_BASE_CLASSES = (pytz.tzinfo.BaseTzInfo, pytz._FixedOffset)
+        if not isinstance(pytz.UTC, pytz._FixedOffset):
+            _PYTZ_BASE_CLASSES = _PYTZ_BASE_CLASSES + (type(pytz.UTC),)
+
+    return isinstance(tz, _PYTZ_BASE_CLASSES)
+
+
+def _datetime_unambiguous(dt, tz):
```

Comment:
This function is mostly used with a `not`. To avoid the double negative, could we swap it and rename it to `_datetime_ambiguous`, so we'd have e.g. `if _datetime_ambiguous(...)` ?

Reply:
We could do this, but there's an unfortunate fuzziness to the terminology here, as "ambiguous" is something of a term of art meaning "this datetime occurs twice in the zone", but this function detects any instance when there are two different possible offsets for a given naïve datetime  — which is to say that "unambiguous" in this context is `not (datetime_ambiguous(dt) or datetime_imaginary(dt))` rather than just `not datetime_ambiguous(dt)`.

I am not quite sure how to resolve this issue with the nomenclature, but I welcome suggestions. 😅 I don't think there's a specific word for "not ambiguous and not imaginary". [PEP 495](https://www.python.org/dev/peps/pep-0495/#terminology) does not seem to suggest a word, but I haven't re-read it carefully lately.

Reply:
I'd be OK with calling the function `_datetime_ambiguous_or_imaginary`.

It's verbose, but anything shorter will be confusing, given what Paul said.

Reply:
OK, I've made that change.